wpe/src/main/cpp/browser/entrypoints.cpp

```diff
@@ -167,6 +163,39 @@ void requestExitFullscreenMode(JNIEnv*, jclass, jint pageId)
     ALOGV("BrowserGlue::requestExitFullscreen(%d) [tid %d]", pageId, gettid());
     Browser::getInstance().requestExitFullscreen(pageId);
 }
+
+void updateAllSettings(JNIEnv* env, jclass, jint pageId, jobject jPageSettings)
+{
+    ALOGV("BrowserGlue::updateAllSettings(%d) [tid %d]", pageId, gettid());
+
+    PageSettings settings;
+
+    jobject obj = env->NewLocalRef(jPageSettings);
+    jclass klass = env->GetObjectClass(obj);
+
+    jmethodID methodID = env->GetMethodID(klass, "getUserAgentString", "()Ljava/lang/String;");
+    if (methodID != nullptr) {
+        jstring jniString = (jstring)env->CallObjectMethod(obj, methodID, 0);
+        const char* str = env->GetStringUTFChars(jniString, 0);
```

Comment:
why would there be an exception? It's just a simple getter which doesn't throw anything.

What do you mean by "You must use nullptr instead of 0 (C++)"? In which case? I think you mean last param in GetStringUTFChars. That's jboolean. Will change that to JNI_FALSE

Reply:
It's true that here you would unlikely have an exception from the Java function itself (if you were calling it from Java), but you may still have an exception from the JVM because it may be in an unknown state, or the class loader has unloaded some dependent classes, or whatever related to stuff that we cannot control from within the native code.

The normal pattern when calling a Java function from native code is always to check for Java exceptions, so you are safe in all use cases.

For the 0, I'm talking about GetStringUTFChars, function signature is `const char* GetStringUTFChars(jstring string, jboolean* isCopy)` so second parameter is a pointer to a jboolean. In C++ it's `nullptr`, if you activate `-Wall` in CFLAGS you will have warnings from the compiler about that.

In C that would be `NULL` which is just a define to 0, but even in this case we would use `NULL` and not 0 because it's a magic number without meaning, which is not good for readbility (https://www.oreilly.com/library/view/c-coding-standards/0321113586/ch18.html)